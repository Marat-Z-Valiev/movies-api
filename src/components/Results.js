import React from "react";
import ResultItem from "./ResultItem";
import ResultsContainerStyled from "./styles/ResultsContainerStyled";
import PropTypes from "prop-types";

const Results = ({totalResults, results}) => {
	const sortedArray = results.sort((a, b) => b.Year - a.Year);
	return (
		<>
			{totalResults ? (
				<p style={{textAlign: "center", color: "#ffffff", fontSize: "1.5rem"}}>
					Total results is {totalResults}
				</p>
			) : (
				""
			)}
			<ResultsContainerStyled>
				{Object.keys(sortedArray).map(key =>
					results[key].Poster === "N/A" ? (
						""
					) : (
						<ResultItem
							className="result-tile"
							key={key}
							result={results[key]}
						/>
					)
				)}
			</ResultsContainerStyled>
		</>
	);
};

Results.propTypes = {
	totalResults: PropTypes.string,
	results: PropTypes.array.isRequired
};

export default Results;
