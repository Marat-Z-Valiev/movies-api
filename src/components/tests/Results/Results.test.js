import React from "react";
import {shallow, configure} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Results from "../../Results";

configure({adapter: new Adapter()});

describe("Results", () => {
	const props = {
		totalResults: "100",
		results: []
	};

	it("should render correctly", () => {
		const wrapper = shallow(<Results {...props} />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
