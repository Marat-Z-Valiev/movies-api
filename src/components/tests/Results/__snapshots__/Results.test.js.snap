// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should render correctly 1`] = `
<Fragment>
  <p
    style={
      Object {
        "color": "#ffffff",
        "fontSize": "1.5rem",
        "textAlign": "center",
      }
    }
  >
    Total results is 
    100
  </p>
  <styled.div />
</Fragment>
`;
